VM/type5_rel2vm_params.py

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env python3
+"""
+VM Parameters Generation
+
+This script generates the velocity model parameters used to generate the velocity model.
+
+Usage
+-----
+To generate VM parameters for a Type-5 realisation:
+
+```
+$ python vm_params_generation.py path/to/realisation.yaml output/vm_params.yaml
+```
+"""
+
+
+from pathlib import Path
+from typing import Annotated, Tuple
+
+import numpy as np
+import scipy as sp
+import shapely
+import typer
+import yaml
+from models import AfshariStewart_2016_Ds, classdef
+from qcore import coordinates
+from shapely import Polygon
+
+from srf_generation import realisation
+from VM import bounding_box
+from VM.bounding_box import BoundingBox
+from VM.models.Bradley_2010_Sa import Bradley_2010_Sa
+
+script_dir = Path(__file__).resolve().parent
+NZ_LAND_OUTLINE = script_dir / "../SrfGen/NHM/res/rough_land.txt"
+
+
+def get_nz_outline_polygon() -> Polygon:
+    """
+    Get the outline polygon of New Zealand.
+
+    Returns:
+        Polygon: The outline polygon of New Zealand.
+    """
+    polygon_coordinates = np.loadtxt(NZ_LAND_OUTLINE)
+    polygon_coordinates = polygon_coordinates[:, ::-1]
+    polygon_coordinates = np.append(
+        polygon_coordinates,
+        np.zeros_like(polygon_coordinates[:, 0]).reshape((-1, 1)),
+        axis=1,
+    )
+    polygon_coordinates = coordinates.wgs_depth_to_nztm(polygon_coordinates)[:, :2]
+    return shapely.Polygon(polygon_coordinates)
+
+
+def pgv_estimate_from_magnitude(magnitude: np.ndarray) -> np.ndarray:
+    """
+    Return PGV for a given magnitude based on default scaling relationship.
+
+    Parameters
+    ----------
+    magnitude : np.ndarray
+        The magnitude(s) of the rupture.
+
+    Returns
+    -------
+    np.ndarray
+        An estimate of PGV for the rupture(s).
+    """
+    return np.interp(
+        magnitude,
+        [3.5, 4.1, 4.7, 5.2, 5.5, 5.8, 6.2, 6.5, 6.8, 7.0, 7.4, 7.7, 8.0],
+        [0.015, 0.0375, 0.075, 0.15, 0.25, 0.4, 0.7, 1.0, 1.35, 1.65, 2.1, 2.5, 3.0],
+    )
+
+
+def find_rrup(magnitude: float, avg_dip: float, avg_rake: float) -> Tuple[float, float]:
+    """
+    Find rrup at which pgv estimated from magnitude is close to target.
```

Comment:
rrup and pgv will probably be understood by anyone using this code but a short description might be informative to some users